src/cli/project/export/conda_explicit_spec.rs

```diff
@@ -0,0 +1,220 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use clap::Parser;
+
+use crate::cli::cli_config::PrefixUpdateConfig;
+use crate::cli::LockFileUsageArgs;
+use crate::lock_file::UpdateLockFileOptions;
+use crate::Project;
+use rattler_conda_types::{ExplicitEnvironmentEntry, ExplicitEnvironmentSpec, Platform};
+use rattler_lock::{CondaPackage, Package, PackageHashes, PypiPackage, PypiPackageData, UrlOrPath};
+
+#[derive(Debug, Parser)]
+#[clap(arg_required_else_help = false)]
+pub struct Args {
+    /// Environment to render
+    #[arg(short, long)]
+    environment: Option<String>,
+
+    /// The platform to render. Defaults to the current platform.
+    #[arg(long)]
+    pub platform: Option<Platform>,
+
+    /// PyPI dependencies are not supported in the conda spec file.
+    /// This flag allows creating the spec file even if PyPI dependencies are present.
+    /// Alternatively see --write-pypi-requirements
+    #[arg(long, default_value = "false")]
+    ignore_pypi_errors: bool,
```

Comment:
Couldnt we just rename this to `--skip-pypi`?

Reply:
I just want to clarify something, since the comment 

> --platform could be implemented but it doesn't have to be in the first version

leads me to believe that there might be some confusion. My understanding is that there are 3 conda formats in the wild currently:

- [`environment.yml`](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-file-manually) This mirrors the manifest in terms of how package versions can be specified and can mix conda and pypi packages. It is platform independent, but supports selectors (at least via `conda-lock`). This can be used by conda/mamba, but requires a full solve. 
- [conda-lock](https://conda.github.io/conda-lock/output/#unified-lockfile) Produced by `conda-lock`. It contains all platform and env information. You can create environments using `conda-lock` or `micromamba`, but not with `conda` itself. 
- [explicit specification file](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#building-identical-conda-environments) This is bound to a single environment/platform combination since it basically just contains raw urls. It is installable directly by conda/mamba and cannot contain pypi packages. 

What this PR contains is an implementation of the *explicit specification file*, so it requires both an environment and platform provided by the user, unless we want to dump a separate file for every combination found in the project. Additionally, since it can't contain pypi packages, a side file must be included or those packages have to be ignored. Making it its own format would require users to make similar choices about what they want in the `requirement.txt` since it can contain loose version bounds or explicit urls (as I'm doing here), and then they would have to run two different export commands.

 I thought to make each format its own subcommand since they will have potentially independent flags, rather than a `--format` or `--kind` (like `conda-lock` uses to switch between formats).

@baszalmstra -- I could change it to `--skip-pypi`, but I originally based my prototype off of `pixi-pack` and `--ignore-pypy-errors` was the flag it was using.

Reply:
Right: ideally at least the docs/interactive `--help` would point at a spec, e.g 
- https://github.com/conda/ceps/pull/81 for `environment.yml`
- https://github.com/conda/ceps/pull/79 for `.txt`/`@EXPLICIT`
  - not sure if there was ever enough consensus on `pip`-compatible `requirements.txt` to have a grammar